File: Python/6kyu/Find the Odd Int/solution.py
# I don't know why I chose to solve this with a dictionary.
# Very unnecessary since we could only count the number of times
# a number appears in the sequence and then check the rest of the
# division by 2. If it's not 0, then it is the odd int. Oh well.

def find_it(seq):
    hash = {}
    
    for x in seq:
        if x in hash:
            hash[x] += 1
        else:
            hash[x] = 1
    
    for y in hash:
        if hash[y] % 2 == 1:
            return y

    return 0

File: PHP/6kyu/Find the Odd Int/solution.php
<?php
function findIt(array $seq) : int
{
    $item = array();
    
    for($i = 0; $i < count($seq); $i++) {
      if(array_key_exists(strval($seq[$i]), $item))
        $item[strval($seq[$i])] += 1;
      else
        $item[strval($seq[$i])] = 1;
    }

    foreach($item as $key => $value){
      if($value % 2 == 1)
        return $key;
    }
    
    return 0;
}
?>